### ISSUE TITLE
FAQ entry: when to call takeView()

### ISSUE THREAD

**Issue by rjrjr** [square/mortar#73]
Archiving this here until we get a real FAQ togther.

tl:dr; If we need them, we effectively have two display phases: initial render (takeView call from onFinishInflate) and an update pass (ping your presenter from onAttachedToWindow)

Standard practic is for a View to call presenter#takeView() in onFinishInflate(). This allows the presenter to set up the view before layout and measurement, and keeps things beautifully synchronous. This is all you need 90% of the time.

But sometimes you'll have transient state that gets clobbered by Android's view hierarchy persistance, typically model state that changed behind the scenes while you were elsewhere in the backstack. When this happens and you're tempted to use View#post to fix them later, stop. Take a breath. And instead ping your presenter from  onAttachedToWindow().
